### DISCUSSION TITLE
How to generate a pydantic model from a sqlalchemy model?

### DISCUSSION THREAD

**Discussion by f0ff886f** [fastapi/fastapi#7866] [Questions] [answered]
### Description

Given a SQLAlchemy declarative model
```py
Base = declarative_base()

class EntityORM(Base):
    __tablename__ = "entities"

    id = Column(Integer, primary_key=True)
    name = Column(String, nullable=False)
    version = Column(Integer, nullable=False)
```
and a pydantic model:
```py
class Entity(BaseModel):
    id: int
    name: str
    version: int
```

How do I go from one to the other, like:

```py
e1 = EntityORM(name="foo", version=42)
e2 = Entity(e1) # exception: TypeError: __init__() takes 1 positional argument but 2 were given
```

Somehow FastAPI is doing it with response_model, how can I do that myself? I'd like to always reason about Pydantic models in my code rather than intersperse ORM objects.

**Comment by dmontagu** [comment, marked as answer]
You have to add `orm_mode` to the config:
```python
class Entity(BaseModel):
    id: int
    name: str
    version: int

    class Config:
        orm_mode = True
```

then you can do

```python
entity_orm = EntityORM(...)
entity = Entity.from_orm(entity_orm)
```
This is *essentially* what FastAPI does, by way of the `validate_model` function from pydantic (if I recall correctly...).

**Comment by f0ff886f** [comment]
Awesome, thank you! I did have the `orm_mode = True` configuration (forgot to type it in) but was definitely *not* using `from_orm`, which is the key. Thanks again!

**Comment by tiangolo** [comment]
Thanks for the help here @dmontagu ! :clap: :bow: 

Thanks for reporting back and closing the issue @f0ff886f  :+1:
